src/commandDetails/coin/transfer.ts

```diff
@@ -0,0 +1,105 @@
+import { container } from '@sapphire/framework';
+import { User } from 'discord.js';
+import {
+  CodeyCommandDetails,
+  CodeyCommandOptionType,
+  CodeyCommandResponseType,
+  SapphireMessageExecuteType,
+  getUserFromMessage
+} from '../../codeyCommand';
+import { getCoinBalanceByUserId, changeDbCoinBalanceByUserId, UserCoinEvent } from '../../components/coin';
+import { getCoinEmoji, getEmojiByName } from '../../components/emojis';
+
+// Update coin balance of a user
+const coinTransferExecuteCommand: SapphireMessageExecuteType = async (client, messageFromUser, args) => {
+  // First mandatory argument is user
+  const user = <User>args['user'];
+  if (!user) {
+    throw new Error('please enter a valid user mention or ID to transfer coins.');
+  }
+
+  // Second mandatory argument is amount
+  const amount = args['amount'];
+  if (typeof amount !== 'number') {
+    throw new Error('please enter a valid amount to transfer.');
+  }
+
+  // Optional argument is reason
+  const reason = args['reason'] ?? null;
+
+  // Get balance
+  const senderBalance = await getCoinBalanceByUserId(getUserFromMessage(messageFromUser).id);
+  const recipientBalance = await getCoinBalanceByUserId(user.id);
+
+  // Implement some basic checks
+  if (messageFromUser.member!.user.id === user.id) {
+    return `You can't transfer money to yourself ${getEmojiByName('codeyConfused')}.`;
+  }
+  if (senderBalance < amount) {
+    return `You don't have enough coins to complete this transfer ${getEmojiByName('codeyConfused')}.`;
+  }
+  if (amount < 0) {
+    return `You can't steal from people ${getEmojiByName('codeyAngry')}.`;
+  }
+  if (amount === 0) {
+    return `What exactly are you trying to do there ${getEmojiByName('codeyStressed')}?`;
+  }
+
+  // Transfer coins
+  await changeDbCoinBalanceByUserId(
```

Comment:
Let's try to use `updateCoinBalanceByUserId` or `adjustCoinBalanceByUserId` instead. These are supposed to be interface functions for coins, since frontend logic shouldn't care about how/coins are stored/updated. As an improvement, we can make `changeDbCoinBalanceByUserId` private.

https://github.com/uwcsc/codeybot/blob/master/src/components/coin.ts#L106